File: simple-stack-example-mvp-fragments/src/main/java/com/zhuinden/simplestackdemoexamplefragments/presentation/paths/taskdetail/TaskDetailKey.java
package com.zhuinden.simplestackdemoexamplefragments.presentation.paths.taskdetail;

import android.support.v4.app.Fragment;
import android.view.View;

import com.google.auto.value.AutoValue;
import com.zhuinden.simplestackdemoexamplefragments.R;
import com.zhuinden.simplestackdemoexamplefragments.application.BaseKey;

/**
 * Created by Zhuinden on 2017.01.25..
 */

@AutoValue
public abstract class TaskDetailKey
        extends BaseKey {
    abstract String taskId();

    public static TaskDetailKey create(String taskId) {
        return new AutoValue_TaskDetailKey(R.layout.path_taskdetail, taskId);
    }

    @Override
    protected Fragment createFragment() {
        return new TaskDetailFragment();
    }

    @Override
    public int menu() {
        return R.menu.taskdetail_fragment_menu;
    }

    @Override
    public boolean isFabVisible() {
        return true;
    }

    @Override
    public int navigationViewId() {
        return 0;
    }

    @Override
    public boolean shouldShowUp() {
        return true;
    }

    @Override
    public View.OnClickListener fabClickListener(Fragment fragment) {
        return v -> {
            TaskDetailFragment coordinator = (TaskDetailFragment) fragment;
            coordinator.editTask();
        };
    }

    @Override
    public int fabDrawableIcon() {
        return R.drawable.ic_edit;
    }
}

File: simple-stack-example-mvp-fragments/src/main/java/com/zhuinden/simplestackdemoexamplefragments/presentation/paths/taskdetail/TaskDetailKey.kt
package com.zhuinden.simplestackdemoexamplefragments.presentation.paths.taskdetail

import android.support.v4.app.Fragment
import android.view.View
import com.zhuinden.simplestackdemoexamplefragments.R
import com.zhuinden.simplestackdemoexamplefragments.application.BaseKey
import kotlinx.android.parcel.Parcelize

/**
 * Created by Zhuinden on 2018. 08. 20.
 */

@Parcelize
data class TaskDetailKey(val taskId: String) : BaseKey() {
    override fun layout(): Int = R.layout.path_taskdetail

    override val isFabVisible: Boolean
        get() = true

    override fun createFragment(): Fragment = TaskDetailFragment()

    override fun menu(): Int = R.menu.taskdetail_fragment_menu

    override fun navigationViewId(): Int = 0

    override fun shouldShowUp(): Boolean = true

    override fun fabClickListener(f: Fragment): View.OnClickListener {
        return View.OnClickListener { v ->
            val fragment = f as TaskDetailFragment
            fragment.editTask()
        }
    }

    override fun fabDrawableIcon(): Int = R.drawable.ic_edit
}
